docs/api.rst

```diff
@@ -189,13 +189,70 @@ It allows you to maintain and update a "state" (aka feedback, aka seed for the n
   	return { seed: values, value: avg }
   }, [], stream)
 
-zipArrayValues :: ((a, b) -> c) -> [a] -> Stream b -> Stream c
+.. _zipArrayValues 
 
-withArrayValues :: [a] -> Stream b -> Stream a
+zipArrayValues
+^^^^^^^^^^^^^^
 
-chain :: (a -> Stream b) -> Stream a -> Stream b
+.. code-block:: haskell
+
+  zipArrayValues :: ((a, b) -> c) -> [a] -> Stream b -> Stream c
+
+Creates a new stream by applying a function ``((a, b) -> c)`` to the latest value of ``Stream b`` with an increasing index of ``[a]`` until ``Stream b`` ends or there are no more values in ``[a]``.::
```

Comment:
I've been trying to avoid starting every combinator description with "Create a new stream ..." (for sources, i.e. constructors, I think I ended up keeping it).  I feel like our theme of FP/immutability is strong enough that we don't need to say it.  Does that seem ok?

If so, how about something like:

"Apply a function to each event and the array value at the respective index."

Then maybe expand on the short description below the ascii diagram with the additional info about the end-of-stream.  I wonder if there is a way to say it somewhat declaratively, i.e. stating the proportion of the stream, rather than talking about "will end when" kind of things. (I know there are other places in the doc where I've written in the "ends when" style ... might try to do a pass and switch those if we feel the declarative style is the way we want to go)

Here's an attempt:

"The resulting stream will contain the same number of events as the input stream, or `array.length` events, whichever is less."

Thoughts?